lib/dialects/mysql/query-generator.js

```diff
@@ -158,6 +159,79 @@ const QueryGenerator = {
     return `ALTER TABLE ${this.quoteTable(tableName)} CHANGE ${attrString.join(', ')};`;
   },
 
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = _.map(this.parseConditionObject(smth.conditions), condition =>
+          `${this.quoteIdentifier(_.first(condition.path))}->>'\$.${_.tail(condition.path).join('.')}' = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      } else if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json functions
+        if (this._checkValidJsonStatement(smth.path)) {
+          str = smth.path;
+        } else {
+          // Also support json dot notation
+          let path = smth.path;
+          let startWithDot = true;
+
+          // Convert .number. to [number].
+          path = path.replace(/\.(\d+)\./g, '[$1].');
+          // Convert .number$ to [number]
+          path = path.replace(/\.(\d+)$/, '[$1]');
```

Comment:
just to make sure, you meant with `.number$` the end of the string, right?

Reply:
Yes, exactly